src/bun.js/bindings/webcrypto/CryptoKeyOKP.cpp

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "CryptoKeyOKP.h"
+
+#if ENABLE(WEB_CRYPTO)
+
+#include "CryptoAlgorithmRegistry.h"
+#include "JsonWebKey.h"
+#include <wtf/text/Base64.h>
+
+namespace WebCore {
+
+static const ASCIILiteral X25519 { "X25519"_s };
+static const ASCIILiteral Ed25519 { "Ed25519"_s };
+
+static constexpr size_t keySizeInBytesFromNamedCurve(CryptoKeyOKP::NamedCurve curve, CryptoKeyType type)
+{
+    switch (curve) {
+    case CryptoKeyOKP::NamedCurve::X25519:
+        return 32;
+    case CryptoKeyOKP::NamedCurve::Ed25519:
+        return type == CryptoKeyType::Private ? 64 : 32;
+    }
+    return 32;
```

Comment:
```suggestion
    switch (curve) {
    case CryptoKeyOKP::NamedCurve::X25519:
    case CryptoKeyOKP::NamedCurve::Ed25519:
        return 32;
    default:
    		// throw not implemented here...
    }
```

Reply:
> ```suggestion
>     switch (curve) {
>     case CryptoKeyOKP::NamedCurve::X25519:
>     case CryptoKeyOKP::NamedCurve::Ed25519:
>         return 32;
>     default:
>     		// throw not implemented here...
>     }
> ```

I think your change is referencing old code, this reverted the change to fix Ed25519. But yeah the default could be set here

Reply:
I don't think I am. Private keys are 32 bytes, not 64. some libraries use the concatenation of priv+pub as the pub which is 64. OpenSSL/BoringSSL is not one of them.

Reply:
Ahh I see what's going on here. This is a divergence in the implementation. Most implementations only keep the private key itself but in BoringSSL implementation, they add a "public key suffix".
From `curve25519.h` in BoringSSL:
> // Note that, unlike RFC 8032's formulation, our private key representation
> // includes a public key suffix to make multiple key signing operations with the
> // same key more efficient. The RFC 8032 private key is referred to in this
> // implementation as the "seed" and is the first 32 bytes of our private key.

So basically internally we can keep the 64 byte form for efficiency but should only expect at import a 32byte key. I'll fix that 👍

Reply:
I see. So I was also partially wrong. Thanks for being patient with me.

Reply:
No problem, you caught the actual error with importing after all 👍  I was just looking at the C defines and didn't even take a look at what the actual spec expects